File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/entity/adme/AdmeMeasuringHoleDepthEntity.java
package com.shmedo.configlibrary.iot.cmd.entity.adme;

import com.shmedo.configlibrary.ble.interfaces.Validater;

import java.lang.reflect.Field;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  1/6/21 <br/>
 * 描述：     生成ADME测量孔深配置参数拼接指令
 */
public class AdmeMeasuringHoleDepthEntity implements Validater {
    private String movementway;//运动方式（0:上拉，1:下放）
    private String motorspeed;//电机速度
    private String movedistance;//运动距离

    public void setMovementway(String movementway) {
        this.movementway = movementway;
    }

    public void setMotorspeed(String motorspeed) {
        this.motorspeed = motorspeed;
    }

    public void setMovedistance(String movedistance) {
        this.movedistance = movedistance;
    }

    @Override
    public void validate() {

    }

    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder();
        try {
            for (Field f : getClass().getDeclaredFields()) {
                Object value = f.get(this);
                if (value != null && !value.equals("NullKey")) {
                    stringBuilder.append(f.getName());
                    stringBuilder.append("=");
                    stringBuilder.append(value);
                    stringBuilder.append("&");
                }
            }
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
        if (stringBuilder.toString().endsWith("&")) {
            stringBuilder.delete(stringBuilder.length() - 1, stringBuilder.length());
        }

        return stringBuilder.toString();
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/iot_cmd/entity/adme/AdmeMeasuringHoleDepthEntity.kt
package com.shmedo.lib.device.base.iot_cmd.entity.adme

import com.shmedo.lib.device.base.Validater

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  1/6/21 <br></br>
 * 描述：     生成ADME测量孔深配置参数拼接指令
 */
class AdmeMeasuringHoleDepthEntity : Validater {
    var movementway //运动方式（0:上拉，1:下放）
            : String? = null
    var motorspeed //电机速度
            : String? = null
    var movedistance //运动距离
            : String? = null

    override fun validate() {}
    override fun toString(): String {
        val stringBuilder = StringBuilder()
        try {
            for (f in javaClass.declaredFields) {
                val value = f[this]
                if (value != null && value != "NullKey") {
                    stringBuilder.append(f.name)
                    stringBuilder.append("=")
                    stringBuilder.append(value)
                    stringBuilder.append("&")
                }
            }
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }
        if (stringBuilder.toString().endsWith("&")) {
            stringBuilder.delete(stringBuilder.length - 1, stringBuilder.length)
        }
        return stringBuilder.toString()
    }
}